### ISSUE TITLE
econf_getStringValue return success with NUL result

### ISSUE THREAD

**Issue by freedge** [openSUSE/libeconf#208]
doc mentions: https://manpages.opensuse.org/Tumbleweed/libeconf-devel/libeconf.3.en.html#econf_err_econf_getStringValue_(econf_file_*_kf,_const_char_*_group,_const_char_*_key,_char_**_result)
```
result A newly allocated string or NULL in error case.
Returns:
econf_err ECONF_SUCCESS or error code
```
but it's possible to have result=NULL and econf_err=ECONF_SUCCESS, for example with the following example.ini file:
```
toto
```

and following code:
```
#include <libeconf.h>

#include <math.h>
#include <stdint.h>
#include <stdio.h>
#include <time.h>
#include <inttypes.h>
#include <assert.h>

int main(void) {
        econf_file *key_file;
        assert(econf_readFile(&key_file, "example.ini", "=", "#") == ECONF_SUCCESS);
        size_t key_number = 0;
        char **keys = NULL;
        assert(econf_getKeys(key_file, NULL, &key_number, &keys) == ECONF_SUCCESS);
        assert(key_number == 1);
        char* val = NULL;
        assert(econf_getStringValue (key_file, NULL, keys[0], &val) == ECONF_SUCCESS);
        printf("val=%X\n", val);
}
```

```
./toto
val=0
```

this was discovered by Fedora users https://discussion.fedoraproject.org/t/sudo-crash-cant-access-sudo-segmentation-fault-core-dumped-fedora-40/117037
as pam now relies on a proper behavior from libeconf since https://github.com/linux-pam/linux-pam/commit/9e4e3e42554fc6344a0b20b42aafb48011e9b2c9#diff-1fd87b83dd11acae1bdfd09ccb3031861d3b2d90c7450b07979c9c4c45d6b798L170

**Comment by thkukuk** [comment]
I agree that the comment is a little bit misleading, but if you don't provide a value, the function can only return `NULL`, so the behavior is correct. It's a difference if you provide no value or an empty value.

**Comment by schubi2** [comment]
I have the same opinion as Thorsten has said.
